src/ports/postgres/modules/convex/madlib_keras.py_in

```diff
@@ -0,0 +1,633 @@
+                                 "DOUBLE PRECISION[]", "TIMESTAMP[]",
+                                 "DOUBLE PRECISION", "DOUBLE PRECISION",
+                                 "DOUBLE PRECISION[]", "DOUBLE PRECISION[]"])
+    plpy.execute(create_output_summary_table, [model_arch_table, model_arch_id,
+                                               "madlib_keras",
+                                               start_training_time, end_training_time,
+                                               source_table, validation_table,
+                                               model, dependent_varname,
+                                               independent_varname, name, description,
+                                               None, version, compile_params,
+                                               fit_params, num_iterations, num_classes,
+                                               aggregate_accuracy[-1],
+                                               aggregate_loss[-1],
+                                               aggregate_accuracy, aggregate_loss,
+                                               aggregate_runtime, final_validation_acc,
+                                               final_validation_loss,
+                                               validation_aggregate_accuracy,
+                                               validation_aggregate_loss])
+
+    create_output_table = plpy.prepare("""
+        CREATE TABLE {0} AS
+        SELECT $1 as model_data""".format(model), ["bytea"])
+    plpy.execute(create_output_table, [model_state])
+
+
+def fit_transition(state, ind_var, dep_var, current_seg_id, num_classes,
+                   all_seg_ids, total_buffers_per_seg, architecture,
+                   compile_params, fit_params, use_gpu, previous_state,
+                   **kwargs):
+
+    """
+
+    :param state:
+    :param ind_var:
+    :param dep_var:
+    :param current_seg_id:
+    :param num_classes:
+    :param all_seg_ids:
+    :param total_buffers_per_seg:
+    :param architecture:
+    :param compile_params:
+    :param fit_params:
+    :param use_gpu:
+    :param previous_state:
+    :param kwargs:
+    :return:
+    """
+    if not ind_var or not dep_var:
+        return state
+
+    start_transition = time.time()
+    SD = kwargs['SD']
+
+    gpus_per_host = 4
+    # Configure GPUs/CPUs
+    device_name = get_device_name_for_keras(
+        use_gpu, current_seg_id, gpus_per_host)
+
+    # Set up system if this is the first buffer on segment'
+    if not state:
+        set_keras_session(use_gpu)
+        segment_model = model_from_json(architecture)
+        compile_and_set_weights(segment_model, compile_params, device_name,
+                                previous_state)
+        SD['segment_model'] = segment_model
+        SD['buffer_count'] = 0
+    else:
+        segment_model = SD['segment_model']
+
+    agg_loss = 0
+    agg_accuracy = 0
```

Comment:
This is the only place in this function where these are set.  Should also be set when `compile_and_set_weights` returns.